packages/expo-updates/ios/EXUpdates/ErrorRecovery.swift

```diff
@@ -308,6 +308,20 @@ public final class ErrorRecovery: NSObject {
   }
 
   @objc private func handleContentDidAppear() {
+    if !RCTIsNewArchEnabled() {
+      self.runContentDidAppearTask()
+      return
+    }
+
+    // In new architecture mode, the `RCTSurfaceHostingProxyRootView` sends the `RCTContentDidAppearNotification` very quick.
```

Comment:
Can you elaborate a bit on this difference? It seems to me like there's probably a lot of things out in the world that rely upon this notification reliably? Is it that in the old architecture it wouldn't send this if there was an exception, but in the new architecture it sends this regardless?

Currently, ErrorRecovery doesn't delay the startup procedure completion so it runs unsynchronized with other procedures, and also there's no updates state communicated to JS about the error recovery. But in the near future, I think it'd be good to change that and emit the `procedureContext.onComplete()` only when the content has appeared (when error recovery is done). Delaying by half a second wouldn't be good for this.

Reply:
Is the issue that RCTContentDidAppearNotification is going out too soon, before the content actually appears? That seems like a bug in RN itself....

Reply:
for the following code, old architecture won't get a chance to send the `RCTContentDidAppearNotification` notification, but new architecture will.

```tsx
import { StyleSheet, Text, View } from 'react-native';

(function() {
  // top level crash
  blah.blah();
})();

export default function App() {
  return (
    <View style={styles.container}>
      <Text>Open up App.js to start working on your app!</Text>
    </View>
  );
}
```

i agree that delay is not a good idea, but i won't say it's a bug from react-native at this moment. because
old arch: RCTRootView -> RCTRootContentView -> RCTContentDidAppearNotification
new arch: RCTSurfaceHostingProxyRootView (no content view) ->  RCTContentDidAppearNotification

let me ask this inconsistent behavior with meta.